src/services/dashboards/dashboard-customize/modules/DashboardAddDefaultWidgetRightArea.vue

```diff
@@ -27,56 +27,115 @@
             </div>
-            <p-json-schema-form v-if="widgetConfig?.widget_options_schema"
-                                :schema="widgetConfig.widget_options_schema"
+            <p-json-schema-form v-if="widgetOptionsJsonSchema"
+                                :schema="widgetOptionsJsonSchema"
                                 :form-data.sync="formData"
                                 class="widget-options-form"
             >
                 <template #label-extra="{ propertyName }">
-                    <div class="inherit-toggle-button">
-                        <span class="text">Inherit</span>
-                        <p-toggle-button :value="inheritFormData[propertyName]"
+                    <div v-if="!nonInheritableProperties.includes(propertyName)"
+                         class="inherit-toggle-button"
+                    >
+                        <span class="text"
+                              :class="{inherit: inheritItemMap[propertyName]}"
+                        >{{ $t('inherit') }}</span><!--song-lang-->
+                        <p-toggle-button :value="inheritItemMap[propertyName]"
                                          @change="handleChangeInheritToggle(propertyName, ...arguments)"
                         />
                     </div>
                 </template>
+                <template #dropdown-extra="{ propertyName, selectedItem }">
+                    <div v-if="isSelected(selectedItem) && inheritItemMap[propertyName]">
+                        <span>{{ selectedItem.label }}</span>
+                        <span class="suffix-text">{{ $t('from dashboard') }}</span><!--song-lang-->
+                    </div>
+                </template>
             </p-json-schema-form>
-            <p-button style-type="secondary"
-                      icon-left="ic_plus_bold"
-                      block
+            <div v-click-outside="hideOptionsMenu"
+                 class="add-options-wrapper"
             >
-                <!--song-lang-->
-                Add Options
-            </p-button>
+                <p-button ref="targetRef"
+                          style-type="secondary"
+                          icon-left="ic_plus_bold"
+                          block
+                          @click="handleClickAddOptions"
+                >
+                    <!--song-lang-->
+                    Add Options
+                </p-button>
+                <p-context-menu v-show="addOptionsMenuVisible"
+                                ref="menuRef"
+                                :menu="optionsMenuItems"
+                                :selected.sync="selectedOptions"
+                                :style="{...contextMenuStyle}"
+                                multi-selectable
+                                item-height-fixed
+                                @select="handleSelectOption"
+                />
```

Comment:
suggestion: Nice :) What about to break it out into a separate component that can be reused in custom dashboard > manage variable button?
It is just an opinion.